components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/UserDefinedFederatedAuthenticatorConfig.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.model;
+
+import org.wso2.carbon.identity.base.AuthenticatorPropertyConstants.DefinedByType;
+
+/**
+ * The user defined federated authenticator configuration model.
+ */
+public class UserDefinedFederatedAuthenticatorConfig extends FederatedAuthenticatorConfig {
```

Comment:
Can't we follow the builder pattern for this ? Or if the only instance variable is endpointConfig we can accept that in the constructor and not have a setter

Reply:
There is another set of attributes inheriting from the FederatedAuthenticatorConfig class.